### ISSUE TITLE
Set up CD for Android (Firebase)

### ISSUE THREAD

**Issue by Tuubz** [nimblehq/ic-flutter-taher-toby#3]
## Why

In order to have convenient testing and debugging, a proper CD must be added.

**Note**: This has already been added in `android_deploy_staging.yml` & `android_deploy_production.yml`.

## Acceptance Criteria

- Create a new application on Firebase
- Add necessary environment variables to secrets
- Every time we merge to `develop`, an Android build has to be deployed to Firebase (Staging)
- Every time we merge to `main`, an Android build has to be deployed to Firebase (Production)

## Design

N/A

## Resources

https://firebase.google.com/

**Comment by Tuubz** [comment]
@luongvo @Shayokh144 

Thanks to `flutter-templates`, I didn't have to add `android_deploy_staging.yml` & `android_deploy_production.yml` myself.

But there's a little bit of set-up left which must be done:

- I created a [Firebase project](https://console.firebase.google.com/project/survey-flutter-ic-5c3bf/overview) for both staging & production 
→ Required for `FIREBASE_ANDROID_APP_ID_STAGING` & `FIREBASE_ANDROID_APP_ID_PRODUCTION`
- I created a [Google Cloud service account](https://console.cloud.google.com/iam-admin/serviceaccounts?project=survey-flutter-ic-5c3bf)
→ Required for `FIREBASE_DISTRIBUTION_CREDENTIAL_JSON`
- I added `FIREBASE_ANDROID_APP_ID_STAGING`, `FIREBASE_ANDROID_APP_ID_PRODUCTION` & `FIREBASE_DISTRIBUTION_CREDENTIAL_JSON` to GitHub's repository secrets

To confirm it works, I created a [test branch](https://github.com/nimblehq/ic-flutter-taher-toby/compare/feature/3-generate-project-from-flutter-templates...chore/2-set-up-cd-for-android?expand=1) 🚀 

In conclusion, there's no pull request needed for this 🙏

**Comment by Tuubz** [comment]
@luongvo Small question, it seems like the following step is redundant in `android_deploy_staging.yml`:

```
- name: Setup .env.staging
    env:
      ENV_STAGING: ${{ secrets.ENV_STAGING }}
    run: |
      echo $ENV_STAGING > .env.staging
```

If I'm not mistaken, `ENV_STAGING` is not being used at all  💭

**Comment by luongvo** [comment]
@Tuubz You will need that to provide your API credentials. If you choose to commit the staging credentials in git, you don't need that step.

**Comment by Tuubz** [comment]
@luongvo Just curious, what's the reason you committed the secrets instead of adding them to GitHub? 💭 

I'm currently running into an issue:

```
SECRET=
REST_API_ENDPOINT=
```

is being parsed to:

```
SECRET=REST_API_ENDPOINT=
```

Did you face the same? 🙏

**Comment by luongvo** [comment]
@Tuubz They were encrypted, mate. I used another approach to this.
- My approach is encrypting & decrypting the committed credential files.
- The support approach from templates is saving the credentials content in env to build the credential files.

**Comment by Tuubz** [comment]
@luongvo Hmmm, but why not add them to GitHub secrets? It looks less complicated than encrypting/decrypting 🙇 

![Screen Shot 2023-03-21 at 14 31 16](https://user-images.githubusercontent.com/18277915/226543783-f080d80d-e03d-4cf5-9dfa-8dc49a382105.png)

**Comment by luongvo** [comment]
@Tuubz it was a long long time ago, mate. The selected approach in templates should be a better choice. I think you have the answer.
